src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -35,7 +147,53 @@ drawIndexedIndirect as it is GPU-validated.
     - range and GPUBuffer are both too small
 `
   )
-  .unimplemented();
+  .params(u =>
+    u
+      .combine('bufferSizeInElements', [10, 100])
+      // Binding size is always no larger than buffer size, make sure that setIndexBuffer succeed
+      .combine('bindingSizeInElements', [10])
+      .combine('drawIndexCount', [10, 11])
+      .combine('drawType', ['drawIndexed', 'drawIndexedIndirect'] as const)
+      .beginSubcases()
+      .combine('indexFormat', ['uint16', 'uint32'] as const)
+      .combine('useBundle', [false, true])
+  )
+  .fn(t => {
+    const {
+      indexFormat,
+      bindingSizeInElements,
+      bufferSizeInElements,
+      drawIndexCount,
+      drawType,
+      useBundle,
+    } = t.params;
+
+    const indexBuffer = t.createIndexBufferWithElementCount(bufferSizeInElements, indexFormat);
+    const indexElementSize = indexFormat === 'uint16' ? 2 : 4;
+    const bindingSize = bindingSizeInElements * indexElementSize;
```

Comment:
`createIndexBufferWithElementCount` helper doesn't seem to be helping here - half of the function is duplicated out here anyway. I think it should be inlined.

Reply:
Done.